<a href="https://colab.research.google.com/github/CodeWithSridhar/Innomatics-Internship/blob/main/Task5_RegEx.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>


# **Task - 5 (RegEx)**

**Questions Link**: https://www.hackerrank.com/domains/python?filters%5Bsubdomains%5D%5B%5D=py-regex 

## Detect Floating Point Number

In [ ]:
import re
class Main():
    def __init__(self):
        self.n = int(input())
        for i in range(self.n):
            self.s = input()
            print(bool(re.match(r'^[-+]?[0-9]*\.[0-9]+$', self.s)))
                    
if __name__ == '__main__':
    Answer = Main()

4
4.0O0
False
-1.00
True
+4.54
True
SomeRandomStuff
False


Explanation:

4.0O0 : O is not a digit.


-1.00 : is valid.

+4.54 : is valid.

SomeRandomStuff: is not a number.

## Group(), Groups() & Groupdict()

In [1]:
import re

expression = r"([a-zA-Z0-9])\1+"
String = re.search(expression,input('Enter string : '))
if String:
    print(String.group(1))
else:
    print(-1)

Enter string : ..12345678910111213141516171820212223
1


Here, .. is the first repeating character, but it is not alphanumeric.
1 is the first (from left to right) alphanumeric repeating character of the string in the substring 111.

## Re.findall() & Re.finditer()

In [2]:
import re
String = re.findall(r'(?<=[qwrtypsdfghjklzxcvbnm])([aeiou]{2,})(?=[qwrtypsdfghjklzxcvbnm])', input().strip(), re.IGNORECASE)

if String:
    for i in String:
        print(i)
else:
    print(-1)

rabcdeefgyYhFjkIoomnpOeorteeeeet
ee
Ioo
Oeo
eeeee


Explanation

ee is located between consonant d and f.

Ioo is located between consonant k and m.

Oeo is located between consonant p and r.

eeeee is located between consonant t and t.